client/src/entrypoints/admin/core.js

```diff
@@ -1,9 +1,15 @@
 import $ from 'jquery';
+
+import { coreControllerDefinitions } from '../../controllers';
 import { escapeHtml } from '../../utils/text';
 import { initButtonSelects } from '../../includes/initButtonSelects';
+import { initStimulus } from '../../includes/stimulus';
 import { initTagField } from '../../includes/initTagField';
 import { initTooltips } from '../../includes/initTooltips';
 
+/** initialise Wagtail Stimulus application with core controller definitions */
+initStimulus({ definitions: coreControllerDefinitions });
```

Comment:
As `core.js` is loaded first I think it would be best that this be included there for now, is it OK to keep it in `core.js`? Assuming we get good progress with cleaning up the core.js my guess is that we would merge the files into one eventually anyway.

As for the DOMContentLoaded - no need - that all happens for us with Stimulus - you can init the application as early or as late as you want it will only start attaching to the DOM when it's ready.

It's actually a really simple Promise approach - see `domReady` and where it's used https://github.com/hotwired/stimulus/blob/main/src/core/application.ts#L110


---

wagtail/admin/templates/wagtailadmin/admin_base.html

```
    <script src="{% versioned_static 'wagtailadmin/js/core.js' %}"></script>
    <script src="{% versioned_static 'wagtailadmin/js/vendor.js' %}"></script>
    <script src="{% versioned_static 'wagtailadmin/js/wagtailadmin.js' %}"></script>
```